lib/rules/space-before-blocks.js

```diff
@@ -33,10 +43,18 @@ module.exports = function(context) {
      */
     function checkPrecedingSpace(node) {
         var precedingToken = context.getTokenBefore(node),
-            hasSpace;
+            hasSpace,
+            parent,
+            requireSpace;
 
         if (precedingToken && !isArrow(precedingToken) && astUtils.isTokenOnSameLine(precedingToken, node)) {
             hasSpace = sourceCode.isSpaceBetweenTokens(precedingToken, node);
+            parent = context.getAncestors().pop();
```

Comment:
You are right, `checkSpaceBeforeCaseBlock` passes a token to `checkPrecedingSpace` and not an `ASTNode`. Would be nice if you rename `node` to `nodeOrToken` to avoid confusion about this in the future.